### DISCUSSION TITLE
Don't match routes that have no element/Component and have children

### DISCUSSION THREAD

**Discussion by corymharper** [remix-run/react-router#10644] [Proposals]
In reading https://github.com/remix-run/react-router/issues/10635#issuecomment-1604907785 I came to understand that this was a design choice for the project, and not a bug. My proposal here is that if a route with children and no element/Component exists, it should not create a matching route at its index that would display nothing. It should only match in the case that a `<Route index />` is defined for that parent route as a child route. This accomplishes two different things in my mind:

1. I can now explicitly display nothing if I want by defining an index route that displays nothing.
2. I won't inadvertently create a matching route where I didn't mean for there to be one.

I can't speak for others, but I primarily use nested routes to group "associated" routes in my application, and to make the best use of relative routes in navigation. The current behavior has caused me to define extra "not found" routes throughout my nested routes to avoid this behavior, and that was only after realizing that these routes would be matched when I hadn't even thought I "defined" them.

**Comment by audunolsen** [comment]
This would make life so much easier for abstractions that generate a route tree. I'm using Vite's `import.meta.glob` and then recursively build a tree, if I don't account for this exact implementation detail of `React Router` then that tree builder function would be **a lot** simpler than it now has to be
